### DISCUSSION TITLE
Enhance SPI using DMA for STM32MP157c-dk2

### DISCUSSION THREAD

**Discussion by swatikumari1-eaton** [zephyrproject-rtos/zephyr#50586] [Q&A]
I want SPI to use DMA so that I can free the processor for tasks.
Enable feature
Update device tree
When sending/receiving data, the processor should be freed up. With just interrupt-based SPI, it seems like the SPI driver is taking a lot of time to the point where Catalina is not able to meet timing requirements for sampling.

I am enabling CONFIG_SPI_STM32_DMA  config then I am getting many linker error

Can somebody suggest!!

**Comment by erwango** [comment]
Not enough information to provide an efficient support.

**Reply by swatikumari1-eaton** [reply]
Want to enable CONFIG_SPI_STM32_DMA for STM32MP157-dk2 Could you suggest how I can proceed.

Thanks!!

**Reply by erwango** [reply]
Please provide more info
What are you doing today ?
What are the errors reported ?

**Reply by swatikumari1-eaton** [reply]
I am trying to enable CONFIG_SPI_STM32_DMA macro in my config file but I am getting below error
![image](https://user-images.githubusercontent.com/96224486/192269701-345c44c7-013a-49b7-a82a-6c9ba1a6df0d.png)

**Reply by erwango** [reply]
ok, this likely requires a fix. Could you raise an issue?

**Reply by swatikumari1-eaton** [reply]
Raised..Thanks!!

**Reply by erwango** [reply]
Ok, Thanks. Note that this isn't a bug as DMA was not supported yet on MP1. We'll work on the support.
